Fix textual DSL (wfconfig editor)

After upgrading to the latest modelserver version (PR #328) , the textual DSL alias the wfconfig file support stopped working.
~~As the modelserver got more strict on loading resources on startup, we have to make sure that all necessary resources can be loaded correctly.  (e.g. check if WorkflowConfiguration needs to be registered as EPackageConfiguration)~~

Update: Please see https://github.com/eclipse-emfcloud/emfcloud-modelserver/issues/71 for details

```
Caused by: java.lang.NoClassDefFoundError: org/eclipse/core/resources/ResourcesPlugin
	at org.eclipse.emf.ecore.plugin.EcorePlugin.getWorkspaceRoot(EcorePlugin.java:1149)
	at org.eclipse.emf.ecore.resource.impl.PlatformResourceURIHandlerImpl.getWorkspaceRoot(PlatformResourceURIHandlerImpl.java:488)
	at org.eclipse.emf.ecore.resource.impl.PlatformResourceURIHandlerImpl.<clinit>(PlatformResourceURIHandlerImpl.java:456)
	at org.eclipse.emf.ecore.resource.URIHandler.<clinit>(URIHandler.java:51)
	at org.eclipse.emf.ecore.resource.impl.ExtensibleURIConverterImpl.<init>(ExtensibleURIConverterImpl.java:138)
	at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.getURIConverter(ResourceSetImpl.java:499)
	at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.getResource(ResourceSetImpl.java:369)
	at org.eclipse.emfcloud.modelserver.common.codecs.XmiCodec.decode(XmiCodec.java:76)
	at org.eclipse.emfcloud.modelserver.common.codecs.XmiCodec.decode(XmiCodec.java:57)
	at org.eclipse.emfcloud.modelserver.common.codecs.XmiCodec.decode(XmiCodec.java:51)
	at org.eclipse.emfcloud.modelserver.client.ModelServerClient.decode(ModelServerClient.java:678)
	at org.eclipse.emfcloud.modelserver.client.ModelServerClient.lambda$2(ModelServerClient.java:114)
	at java.base/java.util.Optional.flatMap(Optional.java:294)
	at org.eclipse.emfcloud.modelserver.client.ModelServerClient.lambda$1(ModelServerClient.java:114)
	at java.base/java.util.function.Function.lambda$andThen$1(Function.java:88)
	at java.base/java.util.function.Function.lambda$andThen$1(Function.java:88)
	at org.eclipse.emfcloud.modelserver.client.Response.body(Response.java:39)
	at com.eclipsesource.coffee.common.ModelServerClientUtil.loadResource(ModelServerClientUtil.java:30)
	at com.eclipsesource.coffee.common.ModelServerClientUtil.loadResource(ModelServerClientUtil.java:34)
	at com.eclipsesource.workflow.dsl.ide.server.WorkflowIndexService.getContent(WorkflowIndexService.java:139)
	at com.eclipsesource.workflow.dsl.ide.server.WorkflowIndexService.workflowFileCreated(WorkflowIndexService.java:119)
```

--- a/README.md
+++ b/README.md
@@ -130,6 +130,7 @@ Use the `com.eclipsesource.coffee.modelserver.CoffeeModelServerLauncher` class t
 #### Coffee GLSP Server
 
 Use the `com.eclipsesource.glsp.example.modelserver.workflow.WorkflowModelServerGLSPServerLauncher` class to start the GLSP Server.
+On the client side, set the `isRunning` flag of the [CoffeeGlspLaunchOptions](web/coffee-server/src/node/backend-module.ts) to `true`.
 
 ### Debug Frontend
 
@@ -138,7 +139,7 @@ Use the `com.eclipsesource.glsp.example.modelserver.workflow.WorkflowModelServer
 
 #### Debug Theia Backend
 
-Use the `Debug Browser Backend` launch config inside VSCode.
+Use the `Launch Browser Backend` launch config inside VSCode.
 
 #### Debug Theia Frontend
 

--- a/backend/plugins/com.eclipsesource.coffee.modelserver/src/main/com/eclipsesource/coffee/modelserver/CoffeeModelServerLauncher.java
+++ b/backend/plugins/com.eclipsesource.coffee.modelserver/src/main/com/eclipsesource/coffee/modelserver/CoffeeModelServerLauncher.java
@@ -11,8 +11,6 @@
 package com.eclipsesource.coffee.modelserver;
 
 import org.apache.commons.cli.ParseException;
-import org.apache.log4j.Level;
-import org.apache.log4j.Logger;
 import org.eclipse.emfcloud.modelserver.emf.di.DefaultModelServerModule;
 import org.eclipse.emfcloud.modelserver.emf.launch.ModelServerLauncher;
 
@@ -22,7 +20,6 @@ public class CoffeeModelServerLauncher {
 
 	public static void main(String[] args) throws ParseException {
 		ModelServerLauncher.configureLogger();
-		Logger.getRootLogger().setLevel(Level.INFO);
 		final ModelServerLauncher launcher = new ModelServerLauncher(args, new DefaultModelServerModule());
 		launcher.addEPackageConfigurations(Lists.newArrayList(CoffeePackageConfiguration.class));
 		launcher.start();

--- a/backend/releng/com.eclipsesource.coffee.target/com.eclipsesource.coffee.target.target
+++ b/backend/releng/com.eclipsesource.coffee.target/com.eclipsesource.coffee.target.target
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" standalone="no"?>
 <?pde?>
 <!-- generated with https://github.com/eclipse-cbi/targetplatform-dsl -->
-<target name="Coffee 2020-09" sequenceNumber="1612516697">
+<target name="Coffee 2020-09" sequenceNumber="1612978832">
   <locations>
     <location includeMode="planner" includeAllPlatforms="false" includeSource="true" includeConfigurePhase="false" type="InstallableUnit">
       <unit id="com.google.gson" version="2.7.0.v20170129-0911"/>
@@ -34,7 +34,7 @@
     <location includeMode="planner" includeAllPlatforms="false" includeSource="true" includeConfigurePhase="false" type="InstallableUnit">
       <unit id="org.eclipse.emfcloud.modelserver.client" version="0.0.0"/>
       <unit id="org.eclipse.emfcloud.modelserver.common" version="0.0.0"/>
-      <repository location="https://download.eclipse.org/emfcloud/modelserver/p2/nightly/0.7/0.7.0.202102041143/"/>
+      <repository location="https://download.eclipse.org/emfcloud/modelserver/p2/nightly/0.7/0.7.0.202102101729/"/>
     </location>
     <location includeMode="planner" includeAllPlatforms="false" includeSource="true" includeConfigurePhase="false" type="InstallableUnit">
       <unit id="org.emfjson.jackson" version="0.0.0"/>

--- a/backend/releng/com.eclipsesource.coffee.target/com.eclipsesource.coffee.target.tpd
+++ b/backend/releng/com.eclipsesource.coffee.target/com.eclipsesource.coffee.target.tpd
@@ -38,7 +38,7 @@ location "http://download.eclipse.org/eclipse/updates/4.17/" eclipse-4.17 {
 location "http://download.eclipse.org/lsp4j/updates/releases/0.9.0" {
 	org.eclipse.lsp4j.sdk.feature.group lazy
 }
-location "https://download.eclipse.org/emfcloud/modelserver/p2/nightly/0.7/0.7.0.202102041143/" {
+location "https://download.eclipse.org/emfcloud/modelserver/p2/nightly/0.7/0.7.0.202102101729/" {
 	org.eclipse.emfcloud.modelserver.client lazy
 	org.eclipse.emfcloud.modelserver.common lazy
 }

--- a/web/.vscode/launch.json
+++ b/web/.vscode/launch.json
@@ -17,7 +17,7 @@
         "--WF_LSP=5017",
         "--WF_ANALYZER=5083",
         "--app-project-path=${workspaceRoot}/browser-app",
-        "--plugins=${workspaceRoot}/browser-app/plugins"
+        "--plugins=local-dir:${workspaceRoot}/browser-app/plugins"
       ],
       "env": {
         "NODE_ENV": "development"

--- a/web/.vscode/settings.json
+++ b/web/.vscode/settings.json
@@ -29,7 +29,6 @@
     "editor.tabSize": 2
   },
   "typescript.tsdk": "node_modules/typescript/lib",
-  "eslint.enable": true,
   "eslint.validate": [
     "javascript",
     "javascriptreact",

--- a/web/.vscode/tasks.json
+++ b/web/.vscode/tasks.json
@@ -29,9 +29,19 @@
       "problemMatcher": []
     },
     {
-      "label": "Coffee: Start Browser Backend",
+      "label": "Coffee: Start Browser Backend along with server JARs",
       "type": "shell",
       "command": "cd browser-app && yarn start",
+      "presentation": {
+        "reveal": "always",
+        "panel": "new"
+      },
+      "problemMatcher": []
+    },
+    {
+      "label": "Coffee: Start Browser Backend (expecting server products to run)",
+      "type": "shell",
+      "command": "cd browser-app && yarn debug",
       "presentation": {
         "reveal": "always",
         "panel": "new"

--- a/web/browser-app/package.json
+++ b/web/browser-app/package.json
@@ -45,8 +45,8 @@
   },
   "scripts": {
     "prepare": "yarn run download:plugins && theia build --mode development",
-    "start": "theia start --WORKFLOW_NOTATION_LSP=5008 --port=3000 --loglevel=debug --root-dir=../../backend/examples/ --loglevel=debug  --plugins=local-dir:./plugins",
-    "debug": "theia start --WORKFLOW_NOTATION_LSP=5008 --WF_LSP=5017 --port=3000 --loglevel=debug --root-dir=../../backend/examples/ --loglevel=debug  --plugins=local-dir:./plugins",
+    "start": "theia start --WORKFLOW_NOTATION_LSP=5008 --port=3000 --root-dir=../../backend/examples/SuperBrewer3000 --plugins=local-dir:./plugins",
+    "debug": "theia start --WORKFLOW_NOTATION_LSP=5008 --WF_LSP=5017 --WF_ANALYZER=5083 --port=3000 --root-dir=../../backend/examples/SuperBrewer3000 --loglevel=debug --plugins=local-dir:./plugins",
     "watch": "theia build --watch --mode development",
     "download:plugins": "theia download:plugins"
   },

--- a/web/coffee-editor-extension/package.json
+++ b/web/coffee-editor-extension/package.json
@@ -36,7 +36,7 @@
     "prepare": "yarn run clean && yarn run build",
     "clean": "rimraf lib",
     "lint": "eslint -c ../.eslintrc.js --ext .ts,.tsx ./src",
-    "build": "tsc",
+    "build": "tsc && yarn run lint",
     "watch": "tsc -w"
   },
   "theiaExtensions": [

--- a/web/coffee-editor-extension/src/browser/coffee-tree/coffee-tree-editor-widget.tsx
+++ b/web/coffee-editor-extension/src/browser/coffee-tree/coffee-tree-editor-widget.tsx
@@ -281,8 +281,7 @@ export class CoffeeTreeEditorWidget extends NavigatableTreeEditorWidget {
                 parentRefSeg === '' ? refToNode : '/' + parentRefSeg + refToNode;
             toCheck = toCheck.parent;
         }
-        const ownerRef = `${this.workspaceService.workspace.uri
-            }/${this.getModelIDToRequest()}#/${refToNode}`;
+        const ownerRef = `${this.workspaceService.workspace.uri}/${this.getModelIDToRequest()}#/${refToNode}`;
         return {
             eClass: node.jsonforms.type,
             $ref: ownerRef.replace('file:///', 'file:/')

--- a/web/coffee-java-extension/package.json
+++ b/web/coffee-java-extension/package.json
@@ -31,7 +31,7 @@
     "prepare": "yarn run clean && yarn run build",
     "clean": "rimraf lib",
     "lint": "eslint -c ../.eslintrc.js --ext .ts,.tsx ./src",
-    "build": "tsc",
+    "build": "tsc && yarn run lint",
     "watch": "tsc -w"
   },
   "theiaExtensions": [

--- a/web/coffee-server/package.json
+++ b/web/coffee-server/package.json
@@ -40,7 +40,7 @@
     "prepare": "yarn run clean && yarn run build",
     "clean": "rimraf lib",
     "lint": "eslint -c ../.eslintrc.js --ext .ts,.tsx ./src",
-    "build": "tsc",
+    "build": "tsc && yarn run lint",
     "watch": "tsc -w",
     "download-server": "cd ./scripts && ./download-server.sh"
   },

--- a/web/coffee-server/src/node/glsp-server-launcher.ts
+++ b/web/coffee-server/src/node/glsp-server-launcher.ts
@@ -39,7 +39,9 @@ export class GLSPServerLauncher implements BackendApplicationContribution {
 
     initialize(): void {
         if (!this.launchOptions.isRunning && !this.start()) {
-            this.logError('Error during model server startup');
+            this.logError('Error during GLSP server startup');
+        } else {
+            this.logInfo('GLSP Server is already running');
         }
     }
 

--- a/web/coffee-welcome-page/package.json
+++ b/web/coffee-welcome-page/package.json
@@ -32,7 +32,8 @@
   "scripts": {
     "prepare": "yarn run clean && yarn run build",
     "clean": "rimraf lib",
-    "build": "tsc",
+    "build": "tsc && yarn run lint",
+    "lint": "eslint -c ../.eslintrc.js --ext .ts,.tsx ./src",
     "watch": "tsc -w"
   },
   "theiaExtensions": [

--- a/web/coffee-welcome-page/src/browser/coffee-welcome-page-contribution.ts
+++ b/web/coffee-welcome-page/src/browser/coffee-welcome-page-contribution.ts
@@ -13,14 +13,18 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  */
-
-import { injectable, inject } from 'inversify';
-import { WelcomePageWidget } from './welcome-page-widget';
-import { CommandRegistry, MenuModelRegistry, MaybePromise } from '@theia/core/lib/common';
-import { CommonMenus, AbstractViewContribution, FrontendApplicationContribution, FrontendApplication } from '@theia/core/lib/browser';
+import {
+    AbstractViewContribution,
+    CommonMenus,
+    FrontendApplication,
+    FrontendApplicationContribution
+} from '@theia/core/lib/browser';
 import { FrontendApplicationStateService } from '@theia/core/lib/browser/frontend-application-state';
-import { WorkspaceService } from '@theia/workspace/lib/browser';
+import { CommandRegistry, MaybePromise, MenuModelRegistry } from '@theia/core/lib/common';
 import { FileNavigatorContribution } from '@theia/navigator/lib/browser/navigator-contribution';
+import { inject, injectable } from 'inversify';
+
+import { WelcomePageWidget } from './welcome-page-widget';
 
 export const WelcomePageCommand = {
     id: WelcomePageWidget.ID,
@@ -31,9 +35,6 @@ export const WelcomePageCommand = {
 export class CoffeeWelcomePageContribution extends AbstractViewContribution<WelcomePageWidget> implements FrontendApplicationContribution {
     @inject(FrontendApplicationStateService)
     protected readonly stateService: FrontendApplicationStateService;
-
-    @inject(WorkspaceService)
-    protected readonly workspaceService: WorkspaceService;
 
     @inject(FileNavigatorContribution)
     protected readonly fileNavigatorContribution: FileNavigatorContribution;
@@ -61,7 +62,7 @@ export class CoffeeWelcomePageContribution extends AbstractViewContribution<Welc
 
     registerCommands(registry: CommandRegistry): void {
         registry.registerCommand(WelcomePageCommand, {
-            execute: () => this.openView({ reveal: true }),
+            execute: () => this.openView({ reveal: true })
         });
     }
 
@@ -69,7 +70,7 @@ export class CoffeeWelcomePageContribution extends AbstractViewContribution<Welc
         menus.registerMenuAction(CommonMenus.HELP, {
             commandId: WelcomePageCommand.id,
             label: WelcomePageCommand.label,
-            order: 'a10'
+            order: 'a11'
         });
     }
 }

--- a/web/coffee-welcome-page/src/browser/coffee-welcome-page-frontend-module.ts
+++ b/web/coffee-welcome-page/src/browser/coffee-welcome-page-frontend-module.ts
@@ -13,26 +13,23 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  */
+import '../../src/browser/style/index.css';
+
+import { bindViewContribution, FrontendApplicationContribution, WidgetFactory } from '@theia/core/lib/browser';
+import { ContainerModule } from 'inversify';
+
+import { CoffeeWelcomePageContribution } from './coffee-welcome-page-contribution';
+import { WelcomePageWidget } from './welcome-page-widget';
 
 /**
  * Generated using theia-extension-generator
  */
-import { CoffeeWelcomePageContribution } from './coffee-welcome-page-contribution';
-import { ContainerModule } from 'inversify';
-import { WelcomePageWidget } from './welcome-page-widget';
-import { WidgetFactory, FrontendApplicationContribution, bindViewContribution } from '@theia/core/lib/browser';
-import { CommandContribution, MenuContribution } from '@theia/core';
-
-import '../../src/browser/style/index.css';
-
 export default new ContainerModule(bind => {
     bindViewContribution(bind, CoffeeWelcomePageContribution);
-    bind(CommandContribution).toService(CoffeeWelcomePageContribution);
-    bind(MenuContribution).toService(CoffeeWelcomePageContribution);
     bind(FrontendApplicationContribution).toService(CoffeeWelcomePageContribution);
     bind(WelcomePageWidget).toSelf();
     bind(WidgetFactory).toDynamicValue(context => ({
         id: WelcomePageWidget.ID,
-        createWidget: () => context.container.get<WelcomePageWidget>(WelcomePageWidget),
+        createWidget: () => context.container.get<WelcomePageWidget>(WelcomePageWidget)
     })).inSingletonScope();
 });

--- a/web/coffee-welcome-page/src/browser/welcome-page-widget.tsx
+++ b/web/coffee-welcome-page/src/browser/welcome-page-widget.tsx
@@ -13,24 +13,25 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
-
-import * as React from 'react';
-import URI from '@theia/core/lib/common/uri';
-import { injectable, inject, postConstruct } from 'inversify';
+import { ApplicationShell, open, OpenerService, SelectableTreeNode, Widget } from '@theia/core/lib/browser';
 import { ReactWidget } from '@theia/core/lib/browser/widgets/react-widget';
 import { CommandRegistry } from '@theia/core/lib/common';
-import { WorkspaceService } from '@theia/workspace/lib/browser';
-import { OpenerService, open, ApplicationShell, SelectableTreeNode } from '@theia/core/lib/browser';
 import { ApplicationInfo, ApplicationServer } from '@theia/core/lib/common/application-protocol';
-import { EXPLORER_VIEW_CONTAINER_ID, FileNavigatorWidget, FILE_NAVIGATOR_ID } from '@theia/navigator/lib/browser';
-import { ANALYZE_COMMAND } from 'coffee-workflow-analyzer/lib/browser/command-contribution';
-import { CODEGEN_COMMAND } from 'coffee-java-extension/lib/browser/command-contribution';
+import URI from '@theia/core/lib/common/uri';
 import { DebugCommands } from '@theia/debug/lib/browser/debug-frontend-application-contribution';
+import { EXPLORER_VIEW_CONTAINER_ID, FILE_NAVIGATOR_ID, FileNavigatorWidget } from '@theia/navigator/lib/browser';
+import { WorkspaceService } from '@theia/workspace/lib/browser';
+import { CODEGEN_COMMAND } from 'coffee-java-extension/lib/browser/command-contribution';
+import { ANALYZE_COMMAND } from 'coffee-workflow-analyzer/lib/browser/command-contribution';
+import { inject, injectable, postConstruct } from 'inversify';
+import * as React from 'react';
+
+/* eslint-disable no-invalid-this */
 
 @injectable()
 export class WelcomePageWidget extends ReactWidget {
 
-    static readonly ID = 'getting.started.widget';
+    static readonly ID = 'welcome.page.widget';
     static readonly LABEL = 'Getting Started';
 
     protected applicationInfo: ApplicationInfo | undefined;
@@ -73,9 +74,9 @@ export class WelcomePageWidget extends ReactWidget {
             <div className='flex-grid'>
                 <div className='col'>
                     {this.renderFeatureSection('Diagram Editor', 'fa fa-project-diagram', (
-                    <p>The example diagram editor allows specifying the behavior of a coffee machine using a flow chart like notation.
-                    The diagram editor is based on <a href='https://www.eclipse.org/glsp/' target='_blank'>the graphical language server platform
-                    (Eclipse GLSP)</a>. Double click the file "superbrewer3000.coffeenotation" in the file explorer or click
+                        <p>The example diagram editor allows specifying the behavior of a coffee machine using a flow chart like notation.
+                    The diagram editor is based on <a href='https://www.eclipse.org/glsp/' target='_blank' rel='noreferrer'>the graphical language server platform
+                    (Eclipse GLSP)</a>. Double click the file &quot;superbrewer3000.coffeenotation&quot; in the file explorer or click
                         the header try out the diagram editor!</p>), this.openDiagram)}
                 </div>
             </div>
@@ -84,8 +85,8 @@ export class WelcomePageWidget extends ReactWidget {
                     {this.renderFeatureSection('Form/Tree Editor', 'fab fa-wpforms', (
                         <p>This editor allows to edit elements in a form-based view along with a tree showing the
                         hierarchy of the model instances. This allows to efficiently browse the model and enter
-                        data. The form editor is based on <a href='https://jsonforms.io' target='_blank'>JSON Forms</a>. Double
-                         click the file "superbrewer3000.coffee" in the file explorer or click the header to try
+                        data. The form editor is based on <a href='https://jsonforms.io' target='_blank' rel='noreferrer'>JSON Forms</a>. Double
+                         click the file &quot;superbrewer3000.coffee&quot; in the file explorer or click the header to try
                           out the editor!</p>), this.openTreeEditor)}
                 </div>
             </div>
@@ -93,46 +94,45 @@ export class WelcomePageWidget extends ReactWidget {
                 <div className='col'>
                     {this.renderFeatureSection('Textual DSL', 'fas fa-indent', (
                         <p>The textual DSL editor allows you to specify model constraints and supports syntax highlighting
-                            and auto completion. It is based on <a href='https://www.eclipse.org/Xtext/' target='_blank'>Xtext</a>. Double click
-                            the file "superbrewer3000.wfconfig" in the file explorer or click the header to try out the textual DSl!'</p>)
-                            , this.openTextualDSL)}
+                            and auto completion. It is based on <a href='https://www.eclipse.org/Xtext/' target='_blank' rel='noreferrer'>Xtext</a>. Double click
+                            the file &quot;superbrewer3000.wfconfig&quot; in the file explorer or click the header to try out the textual DSl!</p>), this.openTextualDSL)}
                 </div>
             </div>
             <div className='flex-grid'>
                 <div className='col'>
                     {this.renderFeatureSection('Model Analysis', 'fas fa-chart-pie', (
-                    <p>Based on the constraints described in the textual DSL, the coffee editor provides an example model analysis.
-                        The result is visualized as a "sun burst" chart. The analysis is an external component written in Kotlin, the
-                        chart is based on D3. Select the file "superbrewer3000.wfconfig" in the file explorer, press F1, type "Analyze
-                        workflow model" and hit enter to see the model analysis in action. Alternatively do a right click in the open
+                        <p>Based on the constraints described in the textual DSL, the coffee editor provides an example model analysis.
+                        The result is visualized as a &quot;sun burst&quot; chart. The analysis is an external component written in Kotlin, the
+                        chart is based on D3. Select the file &quot;superbrewer3000.wfconfig&quot; in the file explorer, press F1, type &quot;Analyze
+                        workflow model&quot; and hit enter to see the model analysis in action. Alternatively do a right click in the open
                         textual DSL editor or click the header above.</p>), this.runModelAnalysis)}
                 </div>
             </div>
             <div className='flex-grid'>
                 <div className='col'>
                     {this.renderFeatureSection('Code Generator', 'fas fa-cogs', (
-                    <p>The coffee editor allows generating example code based on the current model. The code generator itself is written
-                        using Xtend. Right click the file "superbrewer3000.coffee" in the file explorer and select "Generate Workflow code".
-                        Browse the generated code in the "src" and "src-gen" folder, the coffee editor also provides extensive language support
+                        <p>The coffee editor allows generating example code based on the current model. The code generator itself is written
+                        using Xtend. Right click the file &quot;superbrewer3000.coffee&quot; in the file explorer and select &quot;Generate Workflow code&quot;.
+                        Browse the generated code in the &quot;src&quot; and &quot;src-gen&quot; folder, the coffee editor also provides extensive language support
                         for Java!</p>), this.runCodeGenerator)}
                 </div>
             </div>
             <div className='flex-grid'>
                 <div className='col'>
                     {this.renderFeatureSection('Code Editing', 'fab fa-java', (
-                    <p>The coffee editor provides full-fleged Java tooling including syntax highlighting and auto completion.
+                        <p>The coffee editor provides full-fleged Java tooling including syntax highlighting and auto completion.
                         This is based on the Monaco code editor and a Java language server connected via LSP. Make sure you
                         have generated the code first (see above). Then, open any Java file
-                        in the src folder (or click above) and start modifying the code, e.g. by adding "sysout" statements.
+                        in the src folder (or click above) and start modifying the code, e.g. by adding &quot;sysout&quot; statements.
                         </p>), this.openJavaCode)}
                 </div>
             </div>
             <div className='flex-grid'>
                 <div className='col'>
                     {this.renderFeatureSection('Debugging', 'fas fa-bug', (
-                    <p>The coffee editor allows executing and debugging Java code by integrating the debug adapter protocol (DAP).
+                        <p>The coffee editor allows executing and debugging Java code by integrating the debug adapter protocol (DAP).
                         Make sure you have generated the code (see above) and set a break point in any Java file by double
-                        clicking on the left border of the code editor. Press "F5" or click above to start debugging the example.
+                        clicking on the left border of the code editor. Press &quot;F5&quot; or click above to start debugging the example.
                         This will automatically open the integrated debug view and show
                         all outputs of the example code in the console!
                         </p>), this.startDebug)}
@@ -149,11 +149,11 @@ export class WelcomePageWidget extends ReactWidget {
     protected renderHeader(): React.ReactNode {
         return <div className='gs-header'>
             <h1>Coffee Editor <span className='gs-sub-header'>Getting Started</span></h1>
-            <p>The "coffee editor" is a comprehensive example of a web-based modeling tool based on <a href='https://www.eclipse.org/emfcloud/' target='_blank'>EMF.cloud
-            </a> and Eclipse Theia.
+            <p>The &quot;coffee editor&quot; is a comprehensive example of a web-based modeling tool based on&nbsp;
+                <a href='https://www.eclipse.org/emfcloud/' target='_blank' rel='noreferrer'>EMF.cloud</a> and Eclipse Theia.
             Please see the sections below to get an overview of the available features and use the links to directly see them in action.
             Alternatively, <a href='#' onClick={() => this.openFileExplorer()}>open the file explorer</a> to
-            the left and browse the example workspace. See the "Help and more information" section below for further pointers.</p>
+            the left and browse the example workspace. See the &quot;Help and more information&quot; section below for further pointers.</p>
         </div>;
     }
 
@@ -162,9 +162,9 @@ export class WelcomePageWidget extends ReactWidget {
             <a href='#' onClick={opener}>   <h3 className='gs-section-header'>
                 <i className={icon}></i>
                 {title}
-            <span style={{marginLeft: '5px'}}>
-                <i className='fas fa-external-link-alt' />
-            </span>
+                <span style={{ marginLeft: '5px' }}>
+                    <i className='fas fa-external-link-alt' />
+                </span>
             </h3></a>
             {description}
             <div className='gs-action-container'>
@@ -180,16 +180,18 @@ export class WelcomePageWidget extends ReactWidget {
                 Help and more information
             </h3>
             <div className='gs-action-container'>
-                <a href='https://www.eclipse.org/emfcloud/contact/' target='_blank'>Ask a question</a>
+                <a href='https://www.eclipse.org/emfcloud/contact/' target='_blank' rel='noreferrer'>Ask a question</a>
             </div>
             <div className='gs-action-container'>
-                <a href='https://github.com/eclipsesource/coffee-editor/issues' target='_blank'>Report an issue</a>
+                <a href='https://github.com/eclipsesource/coffee-editor/issues' target='_blank' rel='noreferrer'>Report an issue</a>
             </div>
             <div className='gs-action-container'>
-                <a href='https://github.com/eclipsesource/coffee-editor/' target='_blank'>Github project with code and more info</a>
+                <a href='https://github.com/eclipsesource/coffee-editor/' target='_blank' rel='noreferrer'>Github project with code and more info</a>
             </div>
             <div className='gs-action-container'>
-                <a href='https://eclipsesource.com/technology/eclipse-theia/' target='_blank'>Get support for building your own custom tool based on Eclipse Theia</a>
+                <a href='https://eclipsesource.com/technology/eclipse-theia/' target='_blank' rel='noreferrer'>
+                    Get support for building your own custom tool based on Eclipse Theia
+                </a>
             </div>
         </div>;
     }
@@ -202,19 +204,25 @@ export class WelcomePageWidget extends ReactWidget {
         </div>;
     }
 
-    protected openDiagram = () => open(this.openerService, new URI(`${this.workspaceService.workspace?.uri}/superbrewer3000.coffeenotation`));
-    protected openTreeEditor = () => open(this.openerService, new URI(`${this.workspaceService.workspace?.uri}/superbrewer3000.coffee`));
-    protected openTextualDSL = () => open(this.openerService, new URI(`${this.workspaceService.workspace?.uri}/superbrewer3000.wfconfig`));
-    protected openFileExplorer = () => this.shell.revealWidget(EXPLORER_VIEW_CONTAINER_ID);
-    protected runModelAnalysis = () => {
-        open(this.openerService, new URI(`${this.workspaceService.workspace?.uri}/superbrewer3000.wfconfig`))
-            .then(() => { this.commandRegistry.executeCommand(ANALYZE_COMMAND.id); });
-
+    private getSuperBrewer3000FileURI(extension: string): URI {
+        return new URI(`${this.workspaceService.workspace?.uri}/superbrewer3000.${extension}`);
     }
-    protected runCodeGenerator = () => {
+
+    protected openDiagram = (): Promise<object | undefined> => open(this.openerService, this.getSuperBrewer3000FileURI('coffeenotation'));
+    protected openTreeEditor = (): Promise<object | undefined> => open(this.openerService, this.getSuperBrewer3000FileURI('coffee'));
+    protected openTextualDSL = (): Promise<object | undefined> => open(this.openerService, this.getSuperBrewer3000FileURI('wfconfig'));
+    protected openFileExplorer = (): Promise<Widget | undefined> => this.shell.revealWidget(EXPLORER_VIEW_CONTAINER_ID);
+    protected runModelAnalysis = (): void => {
+        open(this.openerService, this.getSuperBrewer3000FileURI('wfconfig'))
+            .then(() => {
+                this.commandRegistry.executeCommand(ANALYZE_COMMAND.id);
+            });
+    };
+
+    protected runCodeGenerator = (): void => {
         this.shell.revealWidget(FILE_NAVIGATOR_ID).then(widget => {
             const { model } = widget as FileNavigatorWidget;
-            const uri = new URI(`${this.workspaceService.workspace?.uri}/superbrewer3000.coffee`);
+            const uri = this.getSuperBrewer3000FileURI('coffee');
             model.revealFile(uri).then(node => {
                 if (SelectableTreeNode.is(node)) {
                     model.selectNode(node);
@@ -222,21 +230,22 @@ export class WelcomePageWidget extends ReactWidget {
                 setTimeout(() => this.commandRegistry.executeCommand(CODEGEN_COMMAND.id, uri), 1000);
             });
         });
+    };
+
+    private getSuperBrewer3000RunnerFileURI(): URI {
+        return new URI(`${this.workspaceService.workspace?.uri}/src/SuperBrewer3000/tests/SuperBrewer3000Runner.java`);
     }
 
-    protected openJavaCode = () => {
+    protected openJavaCode = (): void => {
         this.shell.revealWidget(FILE_NAVIGATOR_ID).then(widget => {
             const { model } = widget as FileNavigatorWidget;
-            const uri = new URI(`${this.workspaceService.workspace?.uri}/src/SuperBrewer3000/tests/SuperBrewer3000Runner.java`);
-            model.revealFile(uri).then(node => {
+            model.revealFile(this.getSuperBrewer3000RunnerFileURI()).then(node => {
                 if (SelectableTreeNode.is(node)) {
                     model.openNode(node);
                 }
             });
-       });
-    }
+        });
+    };
 
-    protected startDebug = () => {
-        this.commandRegistry.executeCommand(DebugCommands.START.id);
-    }
+    protected startDebug = (): Promise<unknown> => this.commandRegistry.executeCommand(DebugCommands.START.id);
 }

--- a/web/coffee-workflow-analyzer-editors/package.json
+++ b/web/coffee-workflow-analyzer-editors/package.json
@@ -29,7 +29,7 @@
     "prepare": "yarn run clean && yarn run build",
     "clean": "rimraf lib",
     "lint": "eslint -c ../.eslintrc.js --ext .ts,.tsx ./src",
-    "build": "tsc",
+    "build": "tsc && yarn run lint",
     "watch": "tsc -w"
   },
   "theiaExtensions": [

--- a/web/coffee-workflow-analyzer-editors/src/node/language-contribution.ts
+++ b/web/coffee-workflow-analyzer-editors/src/node/language-contribution.ts
@@ -21,24 +21,27 @@ export class WorkflowContribution extends BaseLanguageServerContribution {
 
     @inject(ILogger) private readonly logger: ILogger;
 
+    private static readonly ID = 'wfconfig';
+    private static readonly NAME = 'WFCONFIG';
+
     readonly description = {
-        id: this.id,
-        name: this.name,
+        id: WorkflowContribution.ID,
+        name: WorkflowContribution.NAME,
         documentSelector: ['wfconfig'],
         fileEvents: [
             '**/*.wfconfig'
         ]
     };
 
-    readonly id = 'wfconfig';
-    readonly name = 'WFCONFIG';
+    readonly id = WorkflowContribution.ID;
+    readonly name = WorkflowContribution.NAME;
 
     getPort(): number | undefined {
         const arg = process.argv.filter(argument => argument.startsWith('--WF_LSP='))[0];
         if (!arg) {
             return undefined;
         } else {
-            return Number.parseInt(arg.substring('--WF_LSP='.length));
+            return Number.parseInt(arg.substring('--WF_LSP='.length), 10);
         }
     }
 
@@ -51,12 +54,13 @@ export class WorkflowContribution extends BaseLanguageServerContribution {
             const command = 'java';
 
             const jarPath = this.getJarPath();
-            const args: string[] = ['-jar', jarPath];
+            const args: string[] = ['-jar', jarPath, '-startSocket'];
             const serverConnection = await this.createProcessStreamConnectionAsync(command, args);
             this.forward(clientConnection, serverConnection);
         }
     }
-    private async connect(clientConnection: IConnection, socketPort: number) {
+
+    private async connect(clientConnection: IConnection, socketPort: number): Promise<void> {
         const socket = new net.Socket();
         const serverConnection = createSocketConnection(socket, socket, () => {
             this.logger.info('[WorkflowDSL] Socket connection disposed');
@@ -66,7 +70,7 @@ export class WorkflowContribution extends BaseLanguageServerContribution {
         this.forward(clientConnection, serverConnection);
     }
 
-    private getJarPath() {
+    private getJarPath(): string {
         const serverPath = path.resolve(__dirname, '..', '..', 'server');
         const jarPaths = glob.sync('**/plugins/org.eclipse.equinox.launcher_*.jar', { cwd: serverPath });
         if (jarPaths.length === 0) {

--- a/web/coffee-workflow-analyzer/package.json
+++ b/web/coffee-workflow-analyzer/package.json
@@ -34,7 +34,7 @@
     "prepare": "yarn run clean && yarn run build",
     "clean": "rimraf lib",
     "lint": "eslint -c ../.eslintrc.js --ext .ts,.tsx ./src",
-    "build": "tsc",
+    "build": "tsc && yarn run lint",
     "watch": "tsc -w"
   },
   "theiaExtensions": [

--- a/web/coffee-workflow-analyzer/src/node/workflow-analysis-server.ts
+++ b/web/coffee-workflow-analyzer/src/node/workflow-analysis-server.ts
@@ -59,7 +59,7 @@ export class WorkflowAnalysisServer implements WorkflowAnalyzer, BackendApplicat
             return Number.parseInt(arg.substring('--WF_ANALYZER='.length), 10);
         }
     }
-    private getJarPath() {
+    private getJarPath(): string {
         const serverPath = path.resolve(__dirname, '..', '..', 'server');
         const jarPaths = glob.sync('**/plugins/org.eclipse.equinox.launcher_*.jar', { cwd: serverPath });
         if (jarPaths.length === 0) {

